src/renderer/components/+namespaces/namespace-select-filter.tsx

```diff
@@ -0,0 +1,77 @@
+import "./namespace-select.scss";
+
+import React from "react";
+import { observer } from "mobx-react";
+import { components, PlaceholderProps } from "react-select";
+
+import { Icon } from "../icon";
+import { FilterIcon } from "../item-object-list/filter-icon";
+import { FilterType } from "../item-object-list/page-filters.store";
+import { SelectOption } from "../select";
+import { NamespaceSelect } from "./namespace-select";
+import { namespaceStore } from "./namespace.store";
+
+const Placeholder = observer((props: PlaceholderProps<any>) => {
+  const getPlaceholder = (): React.ReactNode => {
+    const namespaces = namespaceStore.contextNamespaces;
+
+    switch (namespaces.length) {
+      case 0:
+      case namespaceStore.allowedNamespaces.length:
+        return <>All namespaces</>;
+      case 1:
+        return <>Namespace: {namespaces[0]}</>;
+      default:
+        return <>Namespaces: {namespaces.join(", ")}</>;
+    }
+  };
+
+  return (
+    <components.Placeholder {...props}>
+      {getPlaceholder()}
+    </components.Placeholder>
+  );
+});
+
+
+@observer
+export class NamespaceSelectFilter extends React.Component {
+  formatOptionLabel({ value: namespace, label }: SelectOption) {
+    if (namespace) {
+      const isSelected = namespaceStore.hasContext(namespace);
+
+      return (
+        <div className="flex gaps align-center">
+          <FilterIcon type={FilterType.NAMESPACE}/>
+          <span>{namespace}</span>
+          {isSelected && <Icon small material="check" className="box right"/>}
+        </div>
+      );
+    }
+
+    return label;
+  }
+
+  onChange([{ value: namespace }]: SelectOption[]) {
```

Comment:
Hi @aleksfront. Could you please explain why do we use SelectOption[] as an array but `toggleContext()` accepts only one string?
 I checked parent `onChange(value: SelectOption, meta: ActionMeta<any>){  if (this.props.onChange) {
      this.props.onChange(value, meta);
    } }` 
    func and it always contains one element array and then it passes this element as a string(if i am not mistaken) to custom `onChange` func().

Reply:
`onChange`'s argument is `SelectOption[]` because we are dealing with an `isMulti={true}` table.

Reply:
@Nokel81 Thanks for describing the issue ^

Reply:
Thank you, guys